### DISCUSSION TITLE
feature: stamina system for making walking great again or: why a deceleration of the gameplay would be a good thing

### DISCUSSION THREAD

**Discussion by Dropgunner** [unitystation/unitystation#7075] [Ideas]
### tl:dr  Limiting running by introducing a stamina system. This would let to improved gameplay 

 I am making the argument that _**constant running has a negative effect on gameplay.**_ 

Unitystation is in itself a fast game. The player has to constantly keep an eye on his environment, read the chat and manage his own tasks. All of this requires high levels of attention

Every factor that speeds up the game contributes to the fact that players cannot enjoy the game because they cannot follow the flow of the game.

They take advantage of the pace of the game to gain an attention advantage over other players. 
Examples are

-  instant plasma flooding, 
-  nukerushes 
- PvP with macros where players have already saved their key combinations in muscle memory and third party tools

 The result are frustrated players who find a destroyed station after 5 minutes. 
I therefore suggest that running in the game should be limited to short or medium sprints. Positive outcomes would be: 

- **Rush strategies will be reduced**, as players will not be able to move as fast as before to execute certain strategies successfully.
- **Players have time to react**, are not overwhelmed by situations, and PVP is less likely to become Clickfest
- **Slower players also mean better latency and less rubber banding.**
- If sprinting was a limited resource, such as through an endurance system, **more tactical battles** would be possible.

**Comment by linked7** [comment]
do you think you could write in paragraphs

**Comment by Dropgunner** [comment]
> do you think you could write in paragraphs

Better?^^

**Comment by Raptorizer** [comment]
As long as stamina is kept too running and only depletes/lowers via running, obesity,  traits,  and leg issues. I am heavily against seeing stamina weaponry that ends up being the go to for combat.

**Comment by linked7** [comment]
moving slow isn't fun

**Comment by voidel** [comment]
Current speeds are pretty comparable to a low/no lag SS13 server.

**Comment by corp-0** [comment]
> Current speeds are pretty comparable to a low/no lag SS13 server.

But you can't deny that we're used to run speed on byond because it most of  the time felt like normal walking speed.

**Comment by daalta** [comment]
You make some very good points.

As someone who's relatively new to the game I've also felt that things happen an a very quick pace - compared to the text-based nature of the game. I wouldn't mind if players weren't *constantly* zipping around.

Stamina also seems very thematic. And funny. I'd love to see players wheeze and collapse on the floor because they've been running nonstop.

Though I feel like a more basic system that forces players to walk after running for X minutes might suffice. So many ways to implement this! Really fun idea 👍

**Comment by NeotericNoble** [comment]
This is supposed to be a faithful recreation of SS13. Now is not the time for balance or design changes. Now is the time for feature parity. Nothing more, nothing less.

**Comment by daalta** [comment]
@NeotericNoble I agree with you in general. Unitystation **is** supposed to be a remake of Unitystation, and we have to be careful when thinking about new features.

Tgstation actually has stamina damage, which is not exhausted by running though. But some Goonstation editions of the game have a stamina system:

> Also located there is the stamina indicator, which governs your endurance. By default, your stamina total is 200 and regenerates at a rate of 10 per tick - click on the indicator to find out about your current status. Every offensive or defensive combat interaction with a player (like attacking or disarming them) will drain stamina. You may also use stamina to move significantly faster by holding the SHIFT key. Once your stamina reaches less than 0 you will be unable to talk, stand up or even interact with any items. Your odds in combat can be improved with a variety of drugs and by wearing protective clothing. 

So to me the idea of a stamina system in Unitystation doesn't seem unfaithful to Space Station 13.

If the idea's fun and makes the game better, then why not implement it?

**Comment by Dropgunner** [comment]
Nobody ever demanded or wrote to remove running in the game. The idea was to make running a resource that can be used tactically and not constantly.  I just checked out SS13 yesterday and found out that it has stuff like janicarts which could be implemented as a general way for transportation in unitystation if someone wants to "run" all the time.

**Comment by daalta** [comment]
@juanototo2
Just to reiterate what @Dropgunner said... This isn't about **removing** running, but making ~~running~~ stamina a resource that replenishes over time.

If we make stamina very generous then it would only affect players who zip around the station non-stop. Combat would be completely unaffected (unless the player just ran a marathon).

If we make stamina a more limited resource, then... "full yakety sax mode" would still be possible, just not forever.

Has a stamina system ever been experimented with? It seems like a touchy topic.
